test/test.js

```diff
@@ -1771,3 +1771,98 @@ test( "Validation triggered on radio and checkbox via click", function() {
 	// test if there is no error anymore
 	ok( form.find( "input.error" ).length === 0, "Form valid" );
 });
+
+test("file accept - image wildcard", function() {
+
+	function dummy() {
+		return file;
+	}
+
+	// https://developer.mozilla.org/en-US/docs/Web/API/FileList
+	var
+		file = {
+		name: "test.png",
+		size: 500001,
+		type: "image/png"
+	},
+		fileList = {
+		0: file,
+		length: 1,
+		item: dummy
+	},
+		input = {},
+		$form = $("<form />"),
+		proxy,
+		validator;
+
+	// dummy DOM element
+	input.type = "file";
+	input.files = fileList;
+	input.nodeName = "INPUT";
+
+	validator = new $.validator({}, $form[0]),
+	proxy = $.proxy($.validator.methods.accept, validator, null, input, "image/*");
+
+	ok( proxy(), "the selected file for upload is valid" );
+});
+
+test("file accept - specified mime type", function() {
+
+	function dummy() {
+		return file;
+	}
+
+	var
+		file = {
+		name: "test.kml",
+		size: 500001,
+		type: "application/vnd.google-earth.kml+xml"
+  }, fileList = {
+		0: file,
+		length: 1,
+		item: dummy
+	},
+		input = {},
+		$form = $("<form />"),
+		validator,
+		proxy;
+
+	input.type = "file";
+	input.files = fileList;
+	input.nodeName = "INPUT";
+
+	validator = new $.validator({}, $form[0]),
+	proxy = $.proxy($.validator.methods.accept, validator, null, input, "application/vnd.google-earth.kml+xml");
+
+	ok( proxy(), "the selected file for upload is valid" );
+});
+
+test("file accept - invalid mime type", function() {
+
+	function dummy() {
+		return file;
+	}
+
+	var file = {
+		name: "test.kml",
+		size: 500001,
+		type: "foobar/vnd.google-earth.kml+xml"
+	}, fileList = {
+		0: file,
+		length: 1,
+		item: dummy
+	},
+		input = {},
+		$form = $("<form />"),
+		validator,
+		proxy;
+
+	input.type = "file";
+	input.files = fileList;
+	input.nodeName = "INPUT";
+
+	validator = new $.validator({}, $form[0]),
+	proxy = $.proxy($.validator.methods.accept, validator, null, input, "application/vnd.google-earth.kml+xml");
+
+	equal(proxy(), "dependency-mismatch", "the selected file for upload is valid" );
```

Comment:
There's a lot of (structural) duplication between these three tests. I think a builder method that returns the necessary object(s) and takes arguments for those parts that are specific for each test would be a good way to refactor this.

Also, since this is testing a method, let's move these tests into `test/methods.js` and name the tests accordingly.